test/common/network/connection_impl_test.cc

```diff
@@ -1068,6 +1080,58 @@ TEST_P(ConnectionImplTest, FlushWriteAndDelayConfigDisabledTest) {
   server_connection->close(ConnectionCloseType::NoFlush);
 }
 
+// Test that tearing down the connection will disable the delayed close timer.
+TEST_P(ConnectionImplTest, DelayedCloseTimeoutDisableOnSocketClose) {
```

Comment:
This might be a big pain, but is it possible to switch all these new tests over to using `InSequence`? For tests like this IMO it makes the tests much stronger.

Reply:
Agreed and done.